src/controllers/handlers/scene-admin-handlers/remove-scene-admin-handler.ts

```diff
@@ -44,26 +48,29 @@ export async function removeSceneAdminHandler(
     throw new InvalidRequestError('Place not found')
   }
 
-  const hasPermission =
-    (isWorlds
-      ? await hasWorldPermission(authenticatedAddress, place.world_name!)
-      : await hasLandPermission(authenticatedAddress, place.positions)) ||
-    (await sceneAdminManager.isAdmin(place.id, authenticatedAddress))
-
-  if (!hasPermission) {
+  const canRemove = await hasPermissionPrivilege(place, authenticatedAddress)
+  if (!canRemove) {
     logger.warn(`User ${authenticatedAddress} is not authorized to remove admins for entity ${place.id}`)
     throw new UnauthorizedError('Only scene admins or the owner can remove admins')
   }
 
-  const isOwnerToRemove = isWorlds
-    ? await hasWorldPermission(adminToRemove, place.world_name!)
-    : await hasLandPermission(adminToRemove, place.positions)
+  const isOwnerToRemove = await isSceneOwner(place, adminToRemove)
 
   if (isOwnerToRemove) {
     logger.warn(`Attempt to remove owner ${adminToRemove} from entity ${place.id} by ${authenticatedAddress}`)
     throw new InvalidRequestError('Cannot remove the owner of the scene')
   }
 
+  const isWorldStreamingPermissionToRemove =
+    place.world && (await hasWorldStreamingPermission(adminToRemove, realmName))
+
+  if (isWorldStreamingPermissionToRemove) {
+    logger.warn(
+      `Attempt to remove world streaming permission ${adminToRemove} from from World Contente Server. Wrong endpoint`
+    )
+    throw new InvalidRequestError('Cannot remove world streaming permission from World Content Server. Wrong endpoint')
```

Comment:
I don't understand this case, would you mind either explaining it here or adding a comment to make it clearer?

Reply:
You can't remove the streaming permission given in world content server